include/iotc.h

```diff
@@ -30,224 +30,192 @@ extern "C" {
 
 /**
  * @brief Shuts down the event engine.
- *
- * @see iotc_events_process_blocking
- * @see iotc_events_process_tick
  */
 extern void iotc_events_stop();
 
 /**
- * @brief Connect to Cloud IoT Core with the provided context.
- *
- * This parameters include a pointer to connection state monitor callback
- * function.
- *
- * <code>iotc_user_callback_t()</code> defines the connection state monitor
- * callback. <code>iotc_user_callback_t()</code> has the following signature:
- *    void foo( iotc_context_handle_t in_context_handle
- *                  , void* data
- *                  , iotc_state_t state )
- *
- * The following are the <code>iotc_user_callback_t()</code> callback function
- * parameters:
- *   - in_context_handle: the context handle provided to
- * <code>iotc_connect()</code>.
- *   - data: a multifunctional structure in the client application. See the <a
- * href="../../../porting_guide.md">porting guide</a> or <code>examples/</code>
- * for more information.
- *   - state: IOTC_STATE_OK if the connection succeeds. See <code><a
- * href="~/include/iotc_error.h">iotc_error.h</a></code> for more error codes.
- *
- * @param [in] iotc_h A context handle created by invoking
- * <code>iotc_create_context()</code>.
+ * @brief Connects to Cloud IoT Core.
+ *
+ * @param [in] iotc_h A {@link iotc_create_context() context handle}.
  * @param [in] username The MQTT username. Cloud IoT Core ignores this
- * parameter.
- * @param [in] password The MQTT password. Cloud IoT Core requires a JWT. Create
- * a JWT with the <code>iotc_create_iotcore_jwt()</code> function.
- * @param [in] client_id The MQTT client identifier. Cloud IoT Core requires a
- *     <a
- * href="https://cloud.google.com/iot/docs/how-tos/mqtt-bridge#device_authentication">device
- * path</a>.
- * @param [in] connection_timeout The number of seconds to keep the socket
- * before CONNACK. If 0, the TCP timeout is used.
- * @param [in] keepalive_timeout The number of seconds that the MQTT service
- * keeps the socket is be kept open without hearing from the client.
- *
- * @see iotc_create_context
- * @see iotc_create_iotcore_jwt
- *
- * @retval IOTC_STATE_OK The connection request is correctly enqueued.
+ *     parameter.
+ * @param [in] password The MQTT password. Cloud IoT Core requires a
+ *     {@link iotc_create_iotcore_jwt() JWT}.
+ * @param [in] client_id The MQTT client ID. Cloud IoT Core requires a
+ *     <a href="https://cloud.google.com/iot/docs/how-tos/mqtt-bridge#device_authentication">
+ *     device path</a>.
+ * @param [in] connection_timeout The number of seconds to wait for a
+ *     <code>CONNACK</code> response before closing the socket. If
+ *     <code>0</code>, the TCP timeout is used.
+ * @param [in] keepalive_timeout The number of seconds the broker will wait for
+ *     the client application to send a message. If no message is sent to the
+ *     broker during the interval, the broker automatically closes the
+ *     connection.
```

Comment:
add
@param [in] client_callback the connection status change callback. The Client SDK will call this function if change happens in the connection status (e.g. if client becomes connected (as a result of this function call) or disconnected for any reasons).
Add this desc to the iotc_connect_to function too, thx!

Reply:
Will do - iotc_user_callback_t specifies the function signature for this?